src/Core/System/StateMachine/StateMachineRegistry.php

```diff
@@ -257,10 +257,6 @@ private function getTransitionDestinationById(string $stateMachineName, string $
         foreach ($stateMachineTransitions as $transition) {
             /** @var StateMachineStateEntity $toState */
             $toState = $transition->getToStateMachineState();
-            // Always allow to cancel a payment whether its a valid transition or not
-            if ($transition->getActionName() === 'cancel' && $transitionName === 'cancel') {
-                return $toState;
-            }
```

Comment:
This is a behaviour break. You need to put this behind a feature flag, `if (Feature::isActive('v6.7.0.0')) {`.

Also, please write a Changelog with a `Next Major version changes` describing that behaviour change.

Reply:
This change has been implemented on our side.